pontoon/sync/changeset.py

```diff
@@ -147,6 +147,12 @@ def update_entity_translations_from_vcs(
             )
         approved_translations = []
 
+        # Unfuzzy existing fuzzy translations to prevent duplicate fuzzies.
+        for translation in db_translations:
```

Comment:
Wouldn't this possibly result in the same translation being in `translations_to_update` multiple times? Perhaps you should do this in the `old_translations` block below.